File: app/src/main/java/com/banew/cw2025_client/ui/main/CreateCoursePlanFragment.java
package com.banew.cw2025_client.ui.main;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.banew.cw2025_backend_common.dto.coursePlans.CoursePlanBasicDto;
import com.banew.cw2025_client.R;

import java.util.ArrayList;
import java.util.List;

public class CreateCoursePlanFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_create_course_plan, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        requireActivity().findViewById(R.id.back_button).setOnClickListener(v -> {
            NavHostFragment.findNavController(this).navigateUp();
        });

        var model = new ViewModelProvider(this).get(CreateCoursePlanView.class);




    }
}

File: app/src/main/java/com/banew/cw2025_client/ui/grave/CreateCoursePlanFragment.kt
package com.banew.cw2025_client.ui.grave

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.NavHostFragment
import com.banew.cw2025_client.R

class CreateCoursePlanFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return super.onCreateView(inflater, container, savedInstanceState)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        requireActivity().findViewById<View?>(R.id.back_button)
            .setOnClickListener(View.OnClickListener { v: View? ->
                NavHostFragment.findNavController(this).navigateUp()
            })

        val model =
            ViewModelProvider(this).get<CreateCoursePlanView>(CreateCoursePlanView::class.java)
    }
}